include/deal.II/lac/constraint_matrix.h

```diff
@@ -1469,16 +1469,23 @@ ConstraintMatrix::add_entry (const size_type line,
   Assert (line != column,
           ExcMessage ("Can't constrain a degree of freedom to itself"));
 
+  // Ensure that the current line is present in the cache:
+  const size_type line_index = calculate_line_index(line);
+  Assert (line_index < lines_cache.size(),
+          ExcMessage("The current ConstraintMatrix does not contain the line "
+                     "for the current entry. Call ConstraintMatrix::add_line "
+                     "before calling this function."));
```

Comment:
It is not clear to me how you can trigger this exception. The `calculate_line_index` function is defined as follows:
```
inline types::global_dof_index
ConstraintMatrix::calculate_line_index (const size_type line) const
{
  //IndexSet is unused (serial case)
  if (!local_lines.size())
    return line;

  Assert(local_lines.is_element(line),
         ExcRowNotStoredHere(line));

  return local_lines.index_within_set(line);
}
```
In other words, this function either throws an exception itself, or its output is within the range `[0,local_lines.n_elements())`. The only possibility where you would trigger this exception is if `lines_cache.size()` is not equal to `local_lines.n_elements()`, but that hopefully never happens...